### DISCUSSION TITLE
fuzz doesn't catch my invariant check

### DISCUSSION THREAD

**Discussion by iftikharuddin** [Cyfrin/security-and-auditing-full-course-s23#19] [Q&A]
This is my `StatelessFuzzCatches` contract, where i set a condition that if `myNumber == 40` it should return zero.

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.20;

// INVARIANT: doMath should never return 0
contract StatelessFuzzCatches {
    /*
     * @dev Should never return 0
     */
    function doMath(uint128 myNumber) public pure returns (uint256) {
        if (myNumber == 40) {
            return 0;
        }
        return 1;
    }
}
```

Here is the Stateless Fuzz test i wrote:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.20;

import {Test} from "forge-std/Test.sol";
import {StatelessFuzzCatches} from "../../src/invariant-break/StatelessFuzzCatches.sol";

contract StatelessFuzzCatchesTest is Test {
    StatelessFuzzCatches public sfc;

    function setUp() public {
        sfc = new StatelessFuzzCatches();
    }

     // Stateless fuzz
     function testFuzzCatchesBugStateless(uint128 randomNumber) public view {
         assert(sfc.doMath(randomNumber) != 0);
     }
}
```

When I run the test `forge test --match-test testFuzzCatchesBugStateless ` the result is below:

```
[PASS] testFuzzCatchesBugStateless(uint128) (runs: 1000, μ: 5717, ~: 5717)
Test result: ok. 1 passed; 0 failed; 0 skipped; finished in 67.28ms
 
Ran 1 test suites: 1 tests passed, 0 failed, 0 skipped (1 total tests)
```

Why it didn't catch the **randomNumber == 40** check? Should I increase the number of runs or am i doing anything incorrect.

**Comment by drained69** [comment]
I think ```fail_on_revert``` was either not added or was set/defaulted to false. You can  add/change this by adding ```fail_on_revert = true``` under the fuzz/invariant section in your toml file.

**Reply by cromewar** [reply]
@iftikharuddin does this make sense to you?

**Reply by iftikharuddin** [reply]
Hey

Thanks for answering, I just changed `fail_on_revert = true` but still same results
<img width="570" alt="image" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/10365184/30044fdb-ec20-466e-98dc-3d64680fc896">

**Contract**:
<img width="545" alt="image" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/10365184/6602c3c5-0833-4e03-bb00-cc087e6e985a">

**Test**:

<img width="640" alt="image" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/10365184/10f7df10-b6fe-44e3-a842-72d296841abc">

**Reply by iftikharuddin** [reply]
Ok so I just changed `runs = 10000` and it worked :) Thanks guys 🙏
